README.md

```diff
@@ -140,6 +140,11 @@ not limited to:
 The `ProtocolError` message must *not* be used to report Sass errors or errors
 running custom functions or importers.
 
+The `ProtocolError` message with `ProtocolErrorType.PARAMS` should only be used
+to report errors for violating any requirements set out by this protocol. Errors
+occured during compliations for violating additional requirements set out by the
+compiler should return `CompileResponse` message with `CompileFailure` instead.
```

Comment:
I think rather than trying to have a generic rubric that implementers have to determine how to apply, we should continue to say "anything that violates a requirement is a PARAMS error by default" and then update the documentation for `CompileRequest.global_functions` to explicitly say that invalid function signatures cause compile errors.

Reply:
Actually the sentence before what I was tying to add says:

> The `ProtocolError` message must *not* be used to report Sass errors or errors
 running custom functions or importers.

Reply:
"Errors running" in that case is intended to mean errors thrown by callbacks passed to the host, not protocol errors involving custom functions or importers.

Reply:
I think other than host directly violating protocol e.g. missing mandatory arguments, the only other case that compiler would throw PARAMS error is for invalid value during deprotoify (e.g. color value out of bound) @nex3 what do you think?

Reply:
That may be true, but I think that's an artifact of the host end of the protocol being fairly reactive. From the compiler end, there are a lot of requirements that aren't just about mandatory paramers. Either way, I think the clearest way forward is to be very explicit for each case where an error should cause a compilation failure.